src/main/sensors/gyro.c

```diff
@@ -98,37 +98,46 @@ typedef struct gyroCalibration_s {
 
 bool firstArmingCalibrationWasStarted = false;
 
+typedef union gyroStage1Filter_u {
+    fastKalman_t fastKalmanFilterState[XYZ_AXIS_COUNT];
+    biquadFilter_t biquadRCFIR2FilterState[XYZ_AXIS_COUNT];
+    pt1Filter_t pt1FilterState[XYZ_AXIS_COUNT];
+} gyroStage1Filter_t;
+
 typedef union gyroSoftFilter_u {
-    biquadFilter_t gyroFilterLpfState[XYZ_AXIS_COUNT];
-    pt1Filter_t gyroFilterPt1State[XYZ_AXIS_COUNT];
-    firFilterDenoise_t gyroDenoiseState[XYZ_AXIS_COUNT];
-} gyroSoftLpfFilter_t;
+    biquadFilter_t biquadFilterState[XYZ_AXIS_COUNT];
+    pt1Filter_t pt1FilterState[XYZ_AXIS_COUNT];
+    firFilterDenoise_t denoiseFilterState[XYZ_AXIS_COUNT];
+} gyroStage2Filter_t;
 
 typedef struct gyroSensor_s {
     gyroDev_t gyroDev;
     gyroCalibration_t calibration;
-    // gyro soft filter
-    filterApplyFnPtr softLpfFilterApplyFn;
-    gyroSoftLpfFilter_t softLpfFilter;
-    filter_t *softLpfFilterPtr[XYZ_AXIS_COUNT];
+
+    // stage1 gyro soft filter (pt1, fkf, biquad rc+fir2)
+    filterApplyFnPtr stage1FilterApplyFn;
+    gyroStage1Filter_t stage1Filter;
+    filter_t *stage1FilterPtr[XYZ_AXIS_COUNT];
```

Comment:
There is no need for this indirection ...

Reply:
@ledvinap this indirection previously existed for lowpass ("stage2"), I just copied the pattern. It seems like a good way to achieve it (as all three of them use separate 'filter state').

What would you suggest instead?

Reply:
For example, it allows me to write like the initialization function to handle all three, like the `lowpass`, this:

```
gyroSensor->stage1FilterPtr[axis] = (filter_t *)&gyroSensor->stage1Filter.fastKalmanFilterState[axis];
gyroSensor->stage1FilterPtr[axis] = (filter_t *)&gyroSensor->stage1Filter.biquadRCFIR2FilterState[axis];
gyroSensor->stage1FilterPtr[axis] = (filter_t *)&gyroSensor->stage1Filter.pt1FilterState[axis];

fastKalmanInit(&gyroSensor->stage1Filter.fastKalmanFilterState[axis], lpfHz, gyroDt);
biquadRCFIR2FilterInit(&gyroSensor->stage1Filter.biquadRCFIR2FilterState[axis], lpfHz, gyroDt);
pt1FilterInit(&gyroSensor->stage1Filter.pt1FilterState[axis], lpfHz, gyroDt);
```